dags/multipod/jax_functional_tests.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run JAX tests."""
+
+import datetime
+from airflow import models
+from dags import composer_env
+from dags.vm_resource import TpuVersion, Zone, Project, V5_NETWORKS, V5P_SUBNETWORKS, RuntimeVersion
+from dags.multipod.configs import jax_tests_gce_config
+from dags.multipod.configs.common import SetupMode
+
+# Run once a day at 10 am UTC (2 am PST)
+SCHEDULED_TIME = "0 10 * * *" if composer_env.is_prod_env() else None
+
+with models.DAG(
+    dag_id="jax_functional_tests",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "jax"],
+    start_date=datetime.datetime(2024, 10, 23),
+    catchup=False,
+) as dag:
+  default_test_name = "jax"
+  test_mode = SetupMode.NIGHTLY
+
+  # v4
+  jax_nightly_1slice_v4_8 = (
+      jax_tests_gce_config.get_jax_distributed_initialize_config(
+          tpu_version=TpuVersion.V4,
+          tpu_cores=8,
+          tpu_zone=Zone.US_CENTRAL2_B.value,
+          time_out_in_min=60,
+          is_tpu_reserved=False,
+          test_name=default_test_name,
+          test_mode=test_mode,
+      )
+  )
+
+  jax_nightly_2slice_v4_8 = (
```

Comment:
Do you want to chain those tests up? This gives us more flexibility (add this at the end, similar to v5p):

`jax_nightly_1slice_v4_8 >> jax_nightly_2slice_v4_8`

1) when you adding more tests in the future, your DAG will not trigger whole tests at one time.

2) when a test in 1 slice, and most likely it will fail at the same reason in 2 slices. Due to the current setting (clean_up step will always be successful, no test after that will be skipped), it will be a little bit hard to skip the 2 slice, but we could keep this logic in mind.

Reply:
Done! Thanks